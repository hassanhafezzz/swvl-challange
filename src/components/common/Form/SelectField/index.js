import React from 'react';
import { useField } from 'formik';

import classNames from 'classnames/bind';
import styles from './styles.module.css';

const cx = classNames.bind(styles);

const SelectField = ({ id, name, label, children }) => {
  const [field, meta] = useField(name);

  return (
    <>
      <label
        htmlFor={id}
        className={cx('label', meta.error && meta.touched ? 'label-error' : '')}
      >
        {label}
      </label>

      <select
        {...field}
        id={id}
        name={name}
        className={cx(
          'select',
          meta.error && meta.touched ? 'error-field' : null,
        )}
      >
        {children}
      </select>

      {meta.error && meta.touched ? (
        <p className={cx('error')}>{meta.error}</p>
      ) : null}
    </>
  );
};

export default SelectField;
