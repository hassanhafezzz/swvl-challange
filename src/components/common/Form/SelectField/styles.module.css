.label {
  margin: 20px 0 8px;
  display: flex;
  text-transform: capitalize;
  font-size: var(--small-paragraph);
  font-weight: var(--bold);
}

.label-error {
  color: var(--color-red);
}

.select {
  cursor: pointer;
  width: 100%;
  height: 45px;
  padding: 10px 15px;
  font-size: var(--paragraph-font);
  color: var(--color-black);
  background-color: var(--color-white);
  border: 2px solid var(--color-gray-100);
  border-radius: 4px;
  box-shadow: none;
  -webkit-appearance: none;
  background-image: url('data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAADIAAAAeCAYAAABuUU38AAAAGXRFWHRTb2Z0d2FyZQBBZG9iZSBJbWFnZVJlYWR5ccllPAAAAUxJREFUeNrM1sEJwkAQBdCsngXPHsQO9O5FS7AAMVYgdqAd2IGCDWgFnryLFQiCZ8EGnJUNimiyM/tnk4HNEAg/8y6ZmMRVqz9eUJvRaSbvutCZ347bXVJy/ZnvTmdJ862Me+hAbZCTs6GHpyUi1tTSvPnqTpoWZPUa7W7ncT3vK4h4zVejy8QzM3WhVUO8ykI6jOxoGA4ig3BLHcNFSCGqGAkig2yqgpEiMsjSfY9LxYQg7L6r0X6wS29YJiYQYecemY+wHrXD1+bklGhpAhBDeu/JfIVGxaAQ9sb8CI+CQSJ+QmJg0Ii/EE2MBiIXooHRQhRCkBhNhBcEhLkwf05ZCG8ICCOpk0MULmvDSY2M8UawIRExLIQIEgHDRoghihgRIgiigBEjgiFATBACAgFgghEwSAAGgoBCBBgYAg5hYKAIFYgHBo6w9RRgAFfy160QuV8NAAAAAElFTkSuQmCC');
  background-size: 12px;
  background-position: calc(100% - 10px) center;
  background-repeat: no-repeat;
}

.select:hover,
.select:focus {
  border: 2px solid var(--color-light-secondary);
  outline: none;
}

.error-field {
  border: 2px solid var(--color-red);
}

.error {
  margin: 5px 0 0;
  display: flex;
  color: var(--color-red);
  font-size: var(--small-paragraph-font);
}
