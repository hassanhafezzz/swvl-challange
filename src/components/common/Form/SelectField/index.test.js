import React from 'react';
import { Formik } from 'formik';
import { render } from '@testing-library/react';

import SelectField from './index';

const component = (
  <Formik>
    <SelectField name="station" id="station" label="station">
      <option disabled defaultValue label="Please select a pick up station" />
      <option value="ahmed oraby" label="ahmed oraby" />
      <option value="almaza" label="almaza" />
    </SelectField>
  </Formik>
);

test('input should have label', () => {
  const { getByLabelText } = render(component);
  const input = getByLabelText('station');
  expect(input).toBeTruthy();
});

test('input snapshot', () => {
  const { container, asFragment } = render(component);
  expect(asFragment(container)).toMatchSnapshot();
});
