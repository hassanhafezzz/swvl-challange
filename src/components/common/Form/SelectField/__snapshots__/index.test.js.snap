// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input snapshot 1`] = `
<DocumentFragment>
  <label
    class="label"
    for="station"
  >
    station
  </label>
  <select
    class="select"
    id="station"
    name="station"
  >
    <option
      disabled=""
      label="Please select a pick up station"
    />
    <option
      label="ahmed oraby"
      value="ahmed oraby"
    />
    <option
      label="almaza"
      value="almaza"
    />
  </select>
</DocumentFragment>
`;
